libnmstate/netapplier.py

```diff
@@ -50,17 +56,19 @@ def apply(
     :rtype: str
     """
     desired_state = copy.deepcopy(desired_state)
-    with plugin_context() as plugin:
+    with plugin_context() as plugins:
         validator.schema_validate(desired_state)
-        validator.validate_capabilities(desired_state, plugin.capabilities)
-        current_state = show_with_plugin(plugin)
+        current_state = show_with_plugins(plugins, include_status_data=True)
+        validator.validate_capabilities(
+            desired_state, plugins_capabilities(plugins)
+        )
         net_state = NetState(desired_state, current_state)
-        checkpoint = plugin.create_checkpoint(rollback_timeout)
-        _apply_ifaces_state(plugin, net_state, verify_change)
+        checkpoints = create_checkpoints(plugins, rollback_timeout)
```

Comment:
I see we need to support multiple checkpoints. Is that possible? How will this affect the user when doing `libnmstate.rollback()`?

Reply:
For checkpoint returned to user, I am encoding it as:


```python
def create_checkpoints(plugins, timeout):
    """
    Return a string containing all the check point created by each plugin in
    the format:
        plugin.name|<checkpoing_path>|plugin.name|<checkpoing_path|...

    """
    checkpoints = []
    for plugin in plugins:
        checkpoint = plugin.create_checkpoint(timeout)
        if checkpoint:
            checkpoints.append(f"{plugin.name}|{checkpoint}")
    return "|".join(checkpoints)
```


For rollback, we ask all plugins to rollback in the order of plugin priority(low priority rollback first).

Reply:
I see. Thanks for explaining!